### PR TITLE
Nekbone fixups and code style alignment

### PR DESCRIPTION
**Pull request by bgeltz**
Incorporates Nekbone changes to only checkout the trunk branch and use the main branch.
Use POSIX compliant function declarations.
Align variable naming with other BASH scripts in the repo.

### CODE REVIEW

File: integration/apps/build_func.sh [hunk 1]
```diff
@@ -48,44 +48,45 @@ function clean_source {
...
     fi
     cd -
 }
 
 # Get the source archive from local cache or web page
-function get_archive {
-    local archive=$1
-    if [ ! -f $archive ]; then
-        if [ -f "$GEOPM_APPS_SRCDIR/$archive" ]; then
-            cp $"$GEOPM_APPS_SRCDIR/$archive" .
```
Comment (bgeltz) [inline integration/apps/build_func.sh:75]: Looks like this was never tested.

File: integration/apps/nekbone/build.sh [hunk 2]
```diff
@@ -32,31 +32,23 @@
...
-    dirname=$1
-    basedir=$PWD
-    tmpdir=$(mktemp -d)
-    cd $tmpdir
-    svn checkout https://repocafe.cels.anl.gov/repos/nekbone
-    mv nekbone/trunk/nekbone $basedir/$dirname
-    cd -
-    rm -rf $tmpdir
+get_nekbone_source() {
+    svn checkout https://repocafe.cels.anl.gov/repos/nekbone/trunk/nekbone ${1}
```
Comment (cmcantalupo) [inline integration/apps/nekbone/build.sh]: The goal with the tmp directory was to get rid of the SVN data.  Be nice to keep that feature.

**Review by cmcantalupo — COMMENTED** [review]
LGTM.  I my rule of thumb is to all caps exported bash variables and down-case local variables.  I'm also not very consistent with that rule.  If you want to all caps everything, I guess that is okay too.  With the tmp checkout my goal was to get rid of the svn data (so we don't have a git repo inside of an svn repo).

File: integration/apps/nekbone/build.sh (hunk 2 above)
Comment (cmcantalupo) [reply integration/apps/nekbone/build.sh]: Also, I've been trying to name $1 in the function with a local variable so it is clear what it is.

**Review by cmcantalupo — APPROVED** [review]

**Pull request merged**